src/commonMain/kotlin/exchange.dydx.abacus/calculator/AccountCalculator.kt

```diff
@@ -19,17 +18,229 @@ class AccountCalculator(val parser: ParserProtocol) {
+            val pending =
+                pendingByMarketId[marketId] as? MutableMap<String, Any>
+            if (pending == null) {
+                pendingByMarketId[marketId] = mutableMapOf(
+                    "firstOrderId" to orderId,
+                    "orderCount" to 1,
+                )
+            } else {
+                pendingByMarketId.safeSet(
+                    "orderCount",
+                    (parser.asInt(pending["orderCount"]) ?: 0) + 1,
+                )
+                pendingByMarketId.safeSet(marketId, pending)
+            }
+        }
+
+        val modifiedPendingPositions = mutableListOf<Any>()
+        for ((marketId, pending) in pendingByMarketId) {
+            val market = parser.asMap(markets?.get(marketId)) ?: continue
+            val assetId = parser.asString(market["assetId"]) ?: continue
+
+            val modifiedPendingPosition = mutableMapOf<String, Any>()
+            modifiedPendingPosition.safeSet("assetId", assetId)
+            modifiedPendingPosition.safeSet(
+                "firstOrderId",
+                parser.value(pending, "firstOrderId"),
+            )
+            modifiedPendingPosition.safeSet(
+                "orderCount",
+                parser.value(pending, "orderCount"),
+            )
+            modifiedPendingPosition.safeSet(
+                "quoteBalance",
+                childSubaccount["quoteBalance"],
+            )
+            modifiedPendingPosition.safeSet(
+                "freeCollateral",
+                childSubaccount["freeCollateral"],
+            )
+            modifiedPendingPosition.safeSet("equity", childSubaccount["equity"])
+            modifiedPendingPositions.add(modifiedPendingPosition)
+        }
+        val modifiedParentSubaccount = parentSubaccount.toMutableMap()
+        modifiedParentSubaccount.safeSet(
+            "pendingPositions",
+            modifiedPendingPositions.sortedWith { pending1, pending2 ->
+                val marketId1 = parser.asString(parser.value(pending1, "assetId"))
+                val marketId2 = parser.asString(parser.value(pending2, "assetId"))
+                if (marketId1 != null && marketId2 != null) {
+                    marketId1.compareTo(marketId2)
+                } else {
+                    0
+                }
+            },
+        )
+        return modifiedParentSubaccount
+    }
+
+    private fun sumEquity(
+        parentSubaccount: Map<String, Any>,
+        childSubaccount: Map<String, Any>
+    ): Map<String, Any> {
+        val modifiedParentSubaccount = parentSubaccount.toMutableMap()
+        modifiedParentSubaccount.safeSet(
+            "equity",
+            sum(
+                parser.asMap(parentSubaccount["equity"]),
+                parser.asMap(childSubaccount["equity"]),
+            ),
+        )
+        return modifiedParentSubaccount
+    }
+
+    private fun sum(value1: Map<String, Any>?, value2: Map<String, Any>?): Map<String, Any> {
```

Comment:
instead of `Map<String, Any>?` would it be possible to use `TradeStatesWithDoubleValues?`

Reply:
I see got it!